precompiles/outposts/osmosis/types.go

```diff
@@ -121,6 +121,10 @@ type RawPacketMetadata struct {
 func (r RawPacketMetadata) Validate() error {
 	osmosisSwap := r.Wasm.Msg.OsmosisSwap
 
+	if r.Wasm.Contract == "" {
```

Comment:
Nit: should we check that it is an Osmosis address also?

Reply:
yes, we should validate that this is a:

- valid Osmosis address (`"osmo"` prefix)
- is a contract (32 bytes according to cosmwasm)

Reply:
I thought about it but I didn't add the check because we are the only allowed to specify the address. But yeah,it's better to add the control!